### ISSUE TITLE
Incompatibility with Roam Studio

### ISSUE THREAD

**Issue by Bo-Feng-1024** [paulovieira/roam-reference-path#20]
Hi,

It seems that this issue #18 has not been solved yet. The screenshot showase the situation. Could you please help me solve it? Thank you!
![image](https://github.com/user-attachments/assets/79ea588e-6498-4a09-aa4c-cd3f92ac2cbe)

**Comment by paulovieira** [comment]
Thanks for the report, and sorry for taking a while to reply.

It seems this problem happens only if the block is a heading block (that is, if it is an H1, H2 or H3). In those cases Roam Studio will add a `margin-top` and `margin-bottom` to a related element, which ends up ruining the height calculation in this plugin.

I've inspected the problem and I think I have a solution.

**Comment by Bo-Feng-1024** [comment]
I looked into this further and opened #23, which should fix it.

Root cause: the connector was positioned from each block's top-left corner plus a fixed top-offset constant, which assumes a normal single-line block. As you diagnosed, for heading blocks Roam Studio adds `margin-top`/`margin-bottom` to `div.controls`, which pushes the bullet down — so the "block-top + constant" calculation no longer lands on the bullet.

The fix stops using those constants and instead measures the actual bullet element (`span.bp3-popover-target`) with `getBoundingClientRect()`, drawing the connector from the parent bullet centre to the child bullet centre, with the corner offset computed against the bullet's real containing block. Wherever Roam Studio shifts the bullet, the measurement follows it, so this also lets the Roam-Studio-specific offset constants be removed entirely. It's essentially the approach you suggested above ("take into account the elements that have the extra height"), done by just measuring the rendered position.

I verified it locally with a case that reproduces the Roam Studio margin (bullet pushed ~24px below the block top): the connector still lands on the bullet centre.

**Comment by paulovieira** [comment]
After a more careful review of this PR I'm a bit confused about the scroll handler. 

To start: can you give me a visual example of the problem? If I remove the the `window.addEventListener('scroll', ...)` everything seems to working ok. I need something to reproduce.

About the implementation: the `scrollRedrawScheduled` variable is false when declared, and the scroll handler has an early return if this variable is false. So `requestAnimationFrame()`  is never called, right?

If I understood correctly, this code doing nothing.

Meanwhile, a PR was submitted to https://github.com/Roam-Research/roam-depot with the other corrections.

**Comment by Bo-Feng-1024** [comment]
Ah, I see you already removed it in ba54865 — that's the right call, it isn't needed.

I tried to reproduce the drift again and couldn't either. After the heading/offset change, the line is a `::before` positioned relative to the bullet's containing block, and its length is the distance between the parent and child bullet centres. In a normal outline both bullets are in the same scroll container, so when you scroll they move together, that distance doesn't change, and the line stays glued to the bullets with no redraw. I confirmed it by scrolling a container 180px with no redraw at all — the line stayed exactly on both bullets (0px gap). So the "drift after scrolling" I originally chased was really the heading-misalignment / reflow problem; once those are fixed there's nothing left for a scroll handler to correct.

To answer your code question directly (for the record): the guard is `if (scrollRedrawScheduled) return`, which bails when the flag is *true* — i.e. "a redraw is already queued this frame". On the first scroll it's still false, so it falls through and does call `requestAnimationFrame`; it's the standard rAF-throttle pattern rather than a no-op. So it did run — it just had no drift to correct, which is the real reason it was pointless.

The only case I can construct where it would matter is an ancestor and a descendant scrolling in separate containers (e.g. sticky headings) — niche enough that commenting it out with the "TO BE CONSIDERED" note is exactly right. If it ever comes up, a guarded version scoped to the edited block would cover it.

Thanks for the careful review, and for merging + pushing the depot PR!

**Comment by paulovieira** [comment]
Great, thanks for the clarification. I was misreading the `if (scrollRedrawScheduled) { return }`  line for some reason. Better name these kind of variables as "isSomething" or "hasWhatever" to avoid these misunderstandings.

So it seems to be working without the scroll handler which is good news. I'll do more stress tests in the next days.

**Comment by Bo-Feng-1024** [comment]
Agreed — a boolean like that reads much better as `isRedrawScheduled` / `hasPendingRedraw`; the bare name is exactly what invited the misread. Moot now that it's gone, but a good habit regardless. Sounds good on the stress tests — if anything turns up, happy to dig in. Thanks again!
